ipld-schema.ipldsch

```diff
@@ -0,0 +1,47 @@
+type UCAN struct {
+  version String
+
+  issuer PublicKey
+  audience PublicKey
+  signature Signature
+
+  capabilities [Capability]
+  -- All proofs are links, however you could still inline proof
+  -- by using CID with identity hashing algorithm
+  proofs [&UCAN]
+  expiration Int
+
+  facts [Fact]
+  nonce optional String
+  notBefore optional Int
+} representation map {
+  field facts default []
+  field proofs default []
+}
+
+
+type Capability struct {
+  with Resource
+  can Ability
+  -- Any additional domain specific details and/or restrictions of the capability
+  caveats { String: Any }
```

Comment:
I have added this field as per https://github.com/ucan-wg/spec/pull/67#pullrequestreview-972879725 which end removing a discussion thread, about naming. I'm quoting relevant comments here to continue naming discussion

From @expede 
> Yup, all agreed on the overall shape 👍 I'm less certain about the word caveats since the enclosed information may include information beyond what are traditionally considered caveats.
>
> In the current draft spec, we use ext to signal "extra" fields, but I'd love to find a more descriptive term. To some degree, this feels a bit like xattrs/xargs, so maybe something as short as x (in the JSON) is sufficient?

Reply:
Originally I was going for "constraints" instead but then I learned that [zcaps](https://w3c-ccg.github.io/zcap-spec/#caveats) use `caveats` instead and according to one of [dictionary definitions](https://www.merriam-webster.com/dictionary/caveat) it is

> an explanation to prevent misinterpretation

which seem to be exactly it, or maybe I'm failing to see what kind of information would fail to fit that definition.

For what it's worth I have also considered `if`, `etc`, `and`, `when` as an alternative names, yet I still like "caveats" better. That said I'm any name will do.